### PR TITLE
Adds the Alert Controller, allowing anything to insert a message to b…

### PR DESCRIPTION
**Pull request by hey-waffles**
…e rendered for the user

### CODE REVIEW

File: src/client/AlertController.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+      alerts: observable,
+      add: action,
+      success: action,
+      clear: action,
+      close: action,
+      removeByKey: action,
+    });
+  }
+
+  public getTopAlerts() {
```
Comment (hey-waffles) [inline src/client/AlertController.ts:34]: Comment

File: src/client/AlertController.ts [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    const result: AlertMessage[] = [];
+    for(let i = 0; i < 3 && i < this.alerts.length; i ++) {
+      result.push(this.alerts[i]);
+    }
+    return result;
+  }
+
+
+  public success(message: string, ttl = DEFAULT_TTL) { this.add(message, AlertType.Success, ttl); }
```
Comment (hey-waffles) [inline src/client/AlertController.ts:43]: Comment. Tedious, but necessary

File: src/client/AlertController.ts [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+  /**
+   * Removes a single alert from the AlertController stack
+   * @param index The index to remove
+   */
+  public close(index: number) {
+    if (!(index >= 0 && index < this.alerts.length)) { return; }
+    this.alerts.splice(index, 1);
+  }
+
+  public removeByKey(key: string) {
```
Comment (hey-waffles) [inline src/client/AlertController.ts:78]: Comment

File: src/components/AlertMessages.tsx [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    alertElements.push(<AlertElement alert={alert} index={index}/>);
+  });
+
+  return (
+    <>
+      <div className={styles.alertContainer}>
+        {alertElements}
+        <TotalAlerts size={AlertController.alerts.length}/>
+      </div>
+      <Formik
```
Comment (hey-waffles) [inline src/components/AlertMessages.tsx:68]: Remove this

**Review by lucyawrey — APPROVED** [review]

**Pull request merged**